Labs/Deep Learning/400 - Stream Analytics and Machine Learning/3 - Predict/readme.md

```diff
@@ -293,8 +293,15 @@ In this exercise, you will modify the Azure Function that you created in the pre
 	        
 	            var config = 
 	            {
-	                userName: databaseUsername,
-	                password: databasePassword,
+	                authentication:
+	                {
+	                    options:
+	                    {
+	                        userName: databaseUsername,
+	                        password: databasePassword
+	                    },
+	                    type: 'default'
+	                },
```

Comment:
This caused a lot of frustration as the original code persistently outputted `ConnectionError: Login failed for user ''.` indicating a blank username. Found the correct syntax [here](https://docs.microsoft.com/en-us/azure/sql-database/sql-database-connect-query-nodejs) under the top hit for "tedious connect to azure sql database". I didn't spend too much time digging for the origin/root cause but my first thought is a tedious package version difference?

Reply:
Yes, I just discovered this one this week as well. It is due to a recent change in tedious as you surmised. I am in the process of updating the lab. There is also a missing step. To connect to the database from Power BI Desktop, you will need to permit your client IP through the firewall on the database server. I will be adding that step.

Reply:
Thanks also!